CRM/Emailfiling/Hook/BuildForm/Setting.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * BuildForm Hook Class for settings page.
+ */
+class CRM_Emailfiling_Hook_BuildForm_Setting {
+
+  /**
+   * Adds enable/disable processing setting to the settings form.
+   *
+   * @param string $formName
+   *   Form Name.
+   * @param CRM_Core_Form $form
+   *   Form Class object.
+   */
+  public function run($formName, CRM_Core_Form &$form) {
+    if (!$this->shouldRun($formName)) {
+      return;
+    }
+
+    $this->addField($form);
+    $this->addTemplate();
+  }
+
+  /**
+   * Checks if hook should run.
+   *
+   * @param string $formName
+   *  Form Name.
+   *
+   * @return bool
+   *   True if hook should run, false otherwise.
+   */
+  private function shouldRun($formName) {
+    if ($formName === 'CRM_Admin_Form_Setting_Smtp') {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Adds new 'Store a copy of sent emails on activity' field to the form.
+   *
+   * @param CRM_Core_Form $form
+   *   Form Class object.
+   */
+  private function addField(CRM_Core_Form &$form) {
+    $fieldName = 'emailfilingIsOutboundProcessingEnabled';
```

Comment:
Can we make this a constant in this class or in another class, so we can make the changes in one place if we need to change the name of this field. And the label too.